sdks/go/pkg/beam/external_test.go

```diff
@@ -0,0 +1,104 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package beam
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime/graphx"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/typex"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/util/reflectx"
+	"github.com/apache/beam/sdks/go/pkg/beam/options/jobopts"
+)
+
+func TestExternalTagged(t *testing.T) {
+	urn := "my::test::urn"
+	payload := []byte("expected_payload")
+	inTag := "inTag"
+	outTag := "outTag"
+
+	p := NewPipeline()
+	s := p.Root()
+
+	col := Impulse(s)
+	ft := typex.New(reflectx.ByteSlice)
+	outMap := ExternalTagged(s, urn, payload, map[string]PCollection{inTag: col}, map[string]FullType{outTag: ft}, true)
+
+	// Validate the in-construction PCollection types for ExternalTagged.
+	if len(outMap) != 1 {
+		t.Fatalf("ExternalTagged = %v, want 1 output with tag %q", outMap, outTag)
+	}
+	out, ok := outMap[outTag]
+	if !ok {
+		t.Fatalf("ExternalTagged = %v, want 1 output with tag %q", outMap, outTag)
+	}
+	if !typex.IsEqual(out.Type(), ft) {
+		t.Fatalf("ExternalTagged()[%q].Type() = %v, want %v", outTag, out.Type(), ft)
+	}
+
+	// Validate the post-construction pipeline protocol buffer contents.
+
+	ctx := context.Background()
+	envUrn := jobopts.GetEnvironmentUrn(ctx)
```

Comment:
It seems strange to use actual flags in a unit test unless they're meant to be set when calling `go test`. If you need the default flags, it seems better to explicitly set these to the default flag values in this function.

There is the situation where, if the default values are changed and it breaks this test then it will be caught, but that still feels more like something for an integration test to catch than a unit test.

Reply:
In this case, you're right a filled in environment isn't necessary. Cleaned up.